internal/ethapi/api.go

```diff
@@ -1662,10 +1659,7 @@ func (s *TransactionAPI) GetRawTransactionByHash(ctx context.Context, hash commo
 		if tx = s.b.GetPoolTransaction(hash); tx != nil {
 			return tx.MarshalBinary()
 		}
-		if err == nil {
-			return nil, nil
-		}
-		return nil, NewTxIndexingError()
+		return nil, err
```

Comment:
It's wrong. Originally if the transaction is not found and background indexing is not running, null is returned.

Now, we return whatever error we have to the RPC consumers. It's a behavioral change.